src/main/java/Group15/View/MuscleSelectionView.java

```diff
@@ -99,37 +99,86 @@ private static ScrollPane createEquipmentSelectorScrollPane() {
+        equipmentSelectorScrollPane.setMinSize(350, 140);
 
         return equipmentSelectorScrollPane;
     }
 
-    private static void createEquipmentCheckboxFunctionality(CheckBox allCheckbox, List<CheckBox> equipmentCheckBoxes) {
-        allCheckbox.setOnAction(_ -> {
-            if (allCheckbox.isSelected()) {
-                MuscleSelectionView.selectedEquipment.clear();
-                for (CheckBox checkbox : equipmentCheckBoxes) {
-                    if (checkbox != allCheckbox) checkbox.setSelected(false);
+    private static void createEquipmentCheckboxFunctionality(Button allButton, Button noneButton, List<CheckBox> equipmentCheckBoxes) {
+        allButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(!selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.add(equipment);
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.remove(equipment);
+
+                }
+            }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(true);
+            }
+        });
+
+        noneButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.remove(equipment);
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.add(equipment);
+
                 }
             }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(false);
+            }
         });
-        for (CheckBox checkbox : equipmentCheckBoxes) {
-            if (checkbox != allCheckbox) {
-                checkbox.setOnAction(_ -> {
-                    if (checkbox.isSelected()) {
-                        allCheckbox.setSelected(false);
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                        if (!MuscleSelectionView.selectedEquipment.contains(Equipment.valueOf(checkbox.getText()))) MuscleSelectionView.selectedEquipment.add(Equipment.valueOf(checkbox.getText()));
-                    } else {
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                    }
-                });
+
+        for (CheckBox checkBox : equipmentCheckBoxes) {
+            checkBox.setOnAction(_ -> {
+                if(checkBox.isSelected()) {
+                    MuscleSelectionView.selectedEquipment.add(fromTextToEquipment(checkBox.getText()));
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.remove(fromTextToEquipment(checkBox.getText()));
+
+                }
+            });
+        }
+    }
+
+    private static Equipment fromTextToEquipment (String checkBoxText) {
```

Comment:
This is in no way shape or form what i meant when i set dont use valueOf(). You are just using your own version, which is slower. It does, however, fix the errors that would otherwise occur. It is still unsafe, if you have too much time on your hands:
https://stackoverflow.com/questions/13726824/javafx-event-triggered-when-selecting-a-check-box